File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.py
sys.set_int_max_str_digits(100000)
class Solution:
    def trailingZeroes(self, n: int) -> int:
        count=0
        fact=1
        for i in range(2,n+1):
            fact*=i
        f=str(fact)
        for i in range(len(f)-1,-1,-1):
            if(f[i]=='0'):
                count+=1
            else:
                break
        return count

File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.java
class Solution {
    public int trailingZeroes(int n) {
        int count=0;
        long fact=1;
        for(int i=2;i<=n;i++)
        {
            fact*=i;
            while(fact%10==0)
            {
                count++;
                fact/=10;
            }
            fact%=10000000000L;
        }
        return count;
    }
}